doc/designs/functions-for-explicitly-fetched-signature-algorithms.md

```diff
@@ -34,98 +34,90 @@ So far, `EVP_PKEY_sign()` and friends are only expected to act on the
+functions for that purpose.  For oneshot functionality, `EVP_PKEY_sign()`
+and `EVP_PKEY_verify()` remain supported, possibly through an alias for the
+application.
 
 [^1]: the term "primitive" is borrowed from [PKCS#1](https://www.rfc-editor.org/rfc/rfc8017#section-5)
 
 ### Making it possible to verify with an early signature
 
 There are newer verifying algorithms that need to receive the signature
 before processing the data.  This is particularly important for streaming
-functionality.  This design proposes a mechanism to accomodate this, which
-is different than the mechanism you get with `EVP_PKEY_verify()`.
+functionality.  This design proposes a mechanism to accomodate this for the
+streaming style functionality.
 
-### API design style
+New public API - API Reference
+------------------------------
 
-Because the *oneshot*, *update* and *final* functionality of all the function
-groups described here are hinged on the corresponding *init* function, a
-minimalistic design is chosen, especially for the provider interface.
-Additional aliases for applications are not part of this design.
+### For limited input size / oneshot signing with `EVP_SIGNATURE`
 
-*This avoids the confusion of mixing up calls to "the right" update function
-(for example) that does the same thing as its sibblings.*
-
-Public API - API Reference
---------------------------
-
-### For signing with `EVP_SIGNATURE`
+``` C
+int EVP_PKEY_sign_init_ex2(EVP_PKEY_CTX *pctx,
+                           EVP_SIGNATURE *algo,
+                           const OSSL_PARAM params[]);
+```
 
-#### Signing a digest
+### For signing a stream with `EVP_SIGNATURE`
 
 ``` C
-int EVP_PKEY_sign_init_for_digest(EVP_PKEY_CTX *pctx,
-                                  EVP_SIGNATURE *algo,
-                                  const OSSL_PARAM params[]);
+int EVP_PKEY_sign_message_init(EVP_PKEY_CTX *pctx,
+                               EVP_SIGNATURE *algo,
+                               const OSSL_PARAM params[]);
+int EVP_PKEY_sign_message_update(EVP_PKEY_CTX *ctx,
+                                 const unsigned char *in,
+                                 size_t inlen);
+int EVP_PKEY_sign_message_final(EVP_PKEY_CTX *ctx,
+                                unsigned char *sig,
+                                size_t *siglen);
 ```
 
-(notably, a fetched "RSA" should probably not accept `RSA_NO_PADDING` when
-initialized with this call)
-
-#### Signing a stream
+### For limited input size / oneshot verification with `EVP_SIGNATURE`
 
 ``` C
-int EVP_PKEY_sign_init_for_message(EVP_PKEY_CTX *pctx,
-                                   EVP_SIGNATURE *algo,
-                                   const OSSL_PARAM params[]);
-int EVP_PKEY_sign_update(EVP_PKEY_CTX *ctx,
-                         const unsigned char *in, size_t inlen);
-int EVP_PKEY_sign_final(EVP_PKEY_CTX *ctx,
-                        unsigned char *sig, size_t *siglen);
+int EVP_PKEY_verify_ex2(EVP_PKEY_CTX *pctx,
```

Comment:
I think you meant `EVP_PKEY_verify_init_ex2` here?

Reply:
Oops.  Yes